src/client.ts

```diff
@@ -167,6 +167,10 @@ export class SubscriptionClient {
       }
 
       this.client.close();
+      this.client.onopen = null;
```

Comment:
Hi,

After this change we started getting the following error in our app:

```ts
[TypeError: Cannot set property 'onopen' of null]

e.close (webpack://[name]/node_modules/subscriptions-transport-ws/dist/client.js:126:33)
close (webpack://[name]/node_modules/subscriptions-transport-ws/dist/client.js:477:22)
```

I believe it is due to some race conditioning where two simultaneous calls are made to `.close()`.

Unfortunately, I don't have a repro for this bug so it's tricky to tell exactly what went wrong. 

The call stack starts in the `.onclose` handler (see https://github.com/apollographql/subscriptions-transport-ws/blob/v0.9.18/src/client.ts#L580-L584). 

Any ideas?

Reply:
@tretne I think this `Cannot set property 'onopen' of null` error happens when close() is called synchronously from within close() in a nested way (my guess - during this.unsubscribeAll() call - the subscription handlers may have some cleanups within themselves which close the the client).

The solution would be to either:
1. have one more check before using this.client on L169 and below, or
2. assign null to this.client early (if it's applicable).

Reply:
Think @dko-slapdash is right -- I'm able to reproduce this pretty easily by loading my app with the network down.
e.g.
1. Host backend locally
2. Point ngrok at the backend
3. Point the client at ngrok

Refresh the app.

A solution is:

```
const client = this.client
this.client = null
// rest of cleanup ...
```

Reply:
https://github.com/apollographql/subscriptions-transport-ws/pull/829